### ISSUE TITLE
slimserver latest version v7.91 may left a xss vulnerability here without fixed.

### ISSUE THREAD

**Issue by d4wner** [LMS-Community/slimserver#175]
Well, I've tried the latest version 7.91 and an older version 7.75.
I found there exist a xss vulnerability.
For example, you can visit a url like this:
```
http://192.168.xxx.xxx:9000/html/ext/resources/images/default/tree/folder.gif''%22%3E%3Cimg%20src=x%20onerror=alert(123)%3E%3C'%22
```
Then you can get a reflect xss vulnerability here，the html page will show us：

```
<HTML><HEAD><TITLE>404 Not Found</TITLE></HEAD>
<BODY>404 Not Found: html/ext/resources/images/default/tree/folder.gif''"><img src=x onerror=alert(123)><'"

</BODY></HTML>
```

Obviously，when you visit the 404 page  with special character，it will not filter completely.

Wish your response！

**Comment by mherger** [comment]
Thanks for this report - it's related/identical to eg. 

https://nvd.nist.gov/nvd.cfm?cvename=CVE-2017-16567
https://www.exploit-db.com/exploits/43122/

https://nvd.nist.gov/nvd.cfm?cvename=CVE-2017-16568
https://www.exploit-db.com/exploits/43123/

The point is: LMS is not supposed to be publicly accessible. If an attacker has access to those input fields, then you're in serious trouble indeed. Therefore make sure LMS is not accessible to the internet. Like you don't open the terminal to the internet either, do you?

**Comment by d4wner** [comment]
Well, thx to your response, but I think once the attacker can make use of the vulnerability , he may just need this opportunity to detect the inner-net app where he can not reach.

**Comment by mherger** [comment]
What I'm saying is that in order to run this attack, the user would already have to access the inner network. Unless the user has deliberately opened the door for the public and exposed his LMS to the internet. But then he really opened to door to much simpler and more dangerous attacks.

**Comment by michaelherger** [comment]
Commits cf09b99 (7.7) and c58f18d (7.9) do address some of these issues. Thanks!

**Comment by mherger** [comment]
More issues fixed in 3f2a4dd (7.7) and f10471d (7.9)
